algorithms/scaling/scaler.py

```diff
@@ -790,14 +793,23 @@ def round_of_outlier_rejection(self):
         """Perform a round of outlier rejection, set a new outliers array."""
         assert self.global_Ih_table is not None
         if self.params.scaling_options.outlier_rejection:
+            Esq_outlier_indices = determine_Esq_outlier_index_arrays(
+                self.global_Ih_table, self.experiment, self.params.scaling_options.emax
+            )[0]
```

Comment:
Moved below to be conditional on `self.params.scaling_options.emax` being non-zero:
```suggestion
            Esq_outlier_indices = determine_Esq_outlier_index_arrays(
                self.global_Ih_table, self.experiment, self.params.scaling_options.emax
            )[0]
```